mail_activity_team/models/mail_activity_mixin.py

```diff
@@ -21,27 +21,13 @@ def _compute_activity_team_user_ids(self):
     def _search_my_activity_date_deadline(self, operator, operand):
         if not self._context.get("team_activities", False):
             return super()._search_my_activity_date_deadline(operator, operand)
-        query = """
-        SELECT res_users_id
-        FROM mail_activity_team_users_rel
-        WHERE mail_activity_team_id IN (
-            SELECT mail_activity_team_id
-            FROM mail_activity_team_users_rel
-            WHERE res_users_id = %(user_id)s)
-        """
-        user = self.env.uid
-        self.env.cr.execute(
-            query,
-            {
-                "user_id": user,
-            },
-        )
-        users = [row[0] for row in self.env.cr.fetchall()]
         activity_ids = self.env["mail.activity"]._search(
             [
                 ("date_deadline", operator, operand),
                 ("res_model", "=", self._name),
-                ("user_id", "in", users),
+                "|",
+                ("user_id", "=", self.env.user.id),
+                ("team_id", "in", self.env.user.activity_team_ids.ids),
```

Comment:
Would you consider to add an index on the `team_id` field on the activity model to improve the performance of this query?